int8 pin
bool level